server/src/test/java/org/elasticsearch/action/admin/indices/validate/query/ValidateQueryResponseTests.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.action.admin.indices.validate.query;
+
+import org.elasticsearch.ElasticsearchException;
+import org.elasticsearch.action.support.DefaultShardOperationFailedException;
+import org.elasticsearch.action.support.broadcast.AbstractBroadcastResponseTestCase;
+import org.elasticsearch.common.Strings;
+import org.elasticsearch.common.xcontent.XContentParser;
+import org.elasticsearch.test.AbstractStreamableXContentTestCase;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+public class ValidateQueryResponseTests extends AbstractBroadcastResponseTestCase<ValidateQueryResponse> {
+
+    protected static ValidateQueryResponse createRandomValidateQueryResponse(
+        int totalShards, int successfulShards, int failedShards, List<DefaultShardOperationFailedException> failures) {
+        boolean valid = failedShards == 0;
+        List<QueryExplanation> queryExplanations = new ArrayList<>(totalShards);
+        for(DefaultShardOperationFailedException failure: failures) {
+            queryExplanations.add(
+                new QueryExplanation(
+                    failure.index(), failure.shardId(), false, failure.reason(), null
+                )
+            );
+        }
+        return new ValidateQueryResponse(
+            valid, queryExplanations, totalShards, successfulShards, failedShards, failures
+        );
+    }
+
+    protected static ValidateQueryResponse createRandomValidateQueryResponse() {
```

Comment:
do these methods need to be protected?

Reply:
Umm no I think. I will make it `private`.